Block/Menu.php

```diff
@@ -99,8 +99,13 @@ public function __construct(
         $this->templateResolver = $templateResolver;
         $this->imageFile = $imageFile;
         $this->escaper = $escaper;
-        $this->submenuTemplate = $this->getMenuTemplate($this->baseSubmenuTemplate);
         $this->setTemplate($this->getMenuTemplate($this->_template));
+
+        $baseSubmenuTemplate = $this->baseSubmenuTemplate;
```

Comment:
Please move this into a private method, I think it's too much logic for a constructor

Reply:
Done